exchanges/btcmarkets/btcmarkets_websocket.go

```diff
@@ -205,3 +228,27 @@ func (b *BTCMarkets) Subscribe(channelToSubscribe wshandler.WebsocketChannelSubs
 	}
 	return b.WebsocketConn.SendMessage(req)
 }
+
+// Login logs in allowing private ws events
+func (b *BTCMarkets) generateAuthSubscriptions() error {
+	var authSub WsAuthSubscribe
+	signTime := strconv.FormatInt(time.Now().UTC().UnixNano()/1000000, 10)
+	strToSign := "/users/self/subscribe" + "\n" + signTime
+	tempSign := crypto.GetHMAC(crypto.HashSHA512,
+		[]byte(strToSign),
+		[]byte(b.API.Credentials.Secret))
+	sign := crypto.Base64Encode(tempSign)
+	markets, err := b.GetMarkets()
+	if err != nil {
+		return err
+	}
+	for x := range markets {
+		authSub.MarketIDs = append(authSub.MarketIDs, markets[x].MarketID)
```

Comment:
Unfortunately, right now we need to seperately subscribe to each marketID in order to track it. So we cannot batch, even though the API supports it.

Happy to work with you later on it